### DISCUSSION TITLE
Setting CONTAINERD_ADDRESS somewhere for nerdctl

### DISCUSSION THREAD

**Discussion by huyz** [rancher-sandbox/rancher-desktop#2552] [Q&A] [answered]
I'm installing Rancher Desktop 1.4.1 for the first time on MacBook ARM and I can't seem to get nerdctl to read the Unix socket address for the dockerd container runtime (except from the command-line flags, which isn't convenient).

Here are my steps:

```shell
brew install docker docker-compose

mkdir -p ~/.docker/cli-plugins
ln -sfn /opt/homebrew/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose

brew install rancher
```

During the initial setup for Rancher Desktop, I get asked whether to use `containerd` or `dockerd`. I select `dockerd`.

Now when I run `nerdctl`, I get:

```
FATA[0000] cannot access containerd socket "/run/k3s/containerd/containerd.sock" (hint: try running with `--address /var/run/docker/containerd/containerd.sock` to connect to Docker-managed containerd): no such file or directory
```

If I add the `--address` flag it works, but it's not quite convenient.

So I try to set the envvar `CONTAINERD_ADDRESS` but that doesn't work (maybe it's limactl's [fault](https://github.com/lima-vm/lima/issues/961) for not passing it through?).

I try several ways to have `address` be read from `nerdctl.toml`: 
- from `~/.config/nerdctl/nerdctl.toml`
- setting `export NERDCTL_TOML=~/.config/nerdctl/nerdctl.toml`
- from `/etc/nerdctl/nerdctl.toml`

Nothing works except setting a shell alias: `alias nerdctl='nerdctl --address /var/run/docker/containerd/containerd.sock'`

What am I doing wrong?

**Comment by jandubois** [comment, marked as answer]
Please use the `docker` cli with the `dockerd` runtime; the `nerdctl` cli is only meant for `containerd`.

Now `dockerd` is running a `containerd` instance internally, but you should not use it directly; you should always go through the docker socket to talk to it.
